File: app/src/main/java/com/lianpos/db/MyApplication.java
package com.lianpos.db;

import android.app.Application;

import io.realm.Realm;
import io.realm.RealmConfiguration;

/**
 * Created by wangshuai on 2017/11/14 0014.
 */

public class MyApplication extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
        Realm.init(this);
        RealmConfiguration config = new RealmConfiguration.Builder().build();
        Realm.setDefaultConfiguration(config);
    }
}

File: app/src/main/java/com/lianpos/db/MyApplication.kt
package com.lianpos.db

import android.app.Application
import android.util.Log
import cn.jpush.android.api.JPushInterface

import io.realm.Realm
import io.realm.RealmConfiguration

/**
 * Created by wangshuai on 2017/11/14 0014.
 */

class MyApplication : Application() {
    private val TAG = "JPush"
    override fun onCreate() {
        Log.d(TAG, "[ExampleApplication] onCreate")
        super.onCreate()
        Realm.init(this)
        val config = RealmConfiguration.Builder().build()
        Realm.setDefaultConfiguration(config)

        JPushInterface.setDebugMode(true)    // 设置开启日志,发布时请关闭日志
        JPushInterface.init(this)            // 初始化 JPush
    }
}